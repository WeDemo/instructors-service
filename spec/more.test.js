import React from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import More from '../client/src/components/more.jsx';

const renderer = require('react-test-renderer');
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

describe('More Component Snapshot', () => {
  const defaultProps = {
    info: {
      courseInfo: [],
      instInfo: {
        inst_name: "Charlie"
      }
    }
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(< More {...defaultProps} />);
  });

  it('renders More component', () => {  
    const MoreTest = (props) => <More {...defaultProps} {...props} />;
    const MoreTestComponent = renderer.create(<MoreTest />).toJSON();
    expect(MoreTestComponent).toMatchSnapshot();
  });

  it('should render a div', () => {  
    expect(wrapper.find('div')).toBeDefined();     
  });
  
  it('should render an instructorCoursesContainer class', () => {
    expect(wrapper.exists('.instructorCoursesContainer')).toEqual(true);
  });

  it('should not render an instructorCoursesHeader div if there are no courses', () => {
    expect(wrapper.exists('.instructorCoursesHeader')).toEqual(false);
  });

  it('should render an instructorCoursesHeader div if there are courses', () => {
    let props = {
      info: {
        courseInfo: [1],
        instInfo: {
          inst_name: "Charlie"
        }
      }
    }
    let wrapper = shallow(< More {...props} />);
    expect(wrapper.exists('.instructorCoursesHeader')).toEqual(true);
  });

});