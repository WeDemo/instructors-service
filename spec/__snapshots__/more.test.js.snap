// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`More Component Snapshot renders More component 1`] = `
<div
  className="instructorCoursesContainer"
/>
`;
